combined_iterator.go

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mysql
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/conduitio-labs/conduit-connector-mysql/common"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+type combinedIterator struct {
+	snapshotIterator common.Iterator
+	cdcIterator      common.Iterator
+
+	currentIterator common.Iterator
+}
+
+type combinedIteratorConfig struct {
+	snapshotConfig snapshotIteratorConfig
+	cdcConfig      cdcIteratorConfig
+}
+
+func newCombinedIterator(
+	ctx context.Context,
+	config combinedIteratorConfig,
```

Comment:
Nitpick: I'd think about making the constructor dumber and simply getting `snapshotIterator` and `cdcIterator` as parameters, the combined iterator can then just use them and let the caller create whatever iterators it wants.